File: app/src/main/java/com/aprivate/sean/ohcg/Calculate.java
package com.aprivate.sean.ohcg;

public class Calculate {
    public static int calculateFirstHand(int playerCount){
       int firstHand = 10;
       if(54 / playerCount < 10){
           firstHand = 54 / playerCount;
       }
       return firstHand;
    }

    public static int calculateWithEstablishedCardsDealt(int playerCount, int handCount){
        int maxPossible = calculateFirstHand(playerCount);
        int handsOneColumn = handCount % 10;
        if(handsOneColumn == 0){
            handsOneColumn = 1;
        }
        return handsOneColumn > maxPossible ? maxPossible : handsOneColumn;
    }

    public static int calculatePointsEarned(int bid, int tricksTaken){
        if(bid != tricksTaken){
            return tricksTaken - 5;
        }
        return tricksTaken + 10;
    }



}

File: app/src/main/java/com/aprivate/sean/ohcg/Calculate.kt
package com.aprivate.sean.ohcg

object Calculate {
    fun calculateFirstHand(playerCount: Int): Int {
        var firstHand = 10
        if (54 / playerCount < 10) {
            firstHand = 54 / playerCount
        }
        return firstHand
    }

    fun calculateWithEstablishedCardsDealt(playerCount: Int, handCount: Int): Int {
        val maxPossible = calculateFirstHand(playerCount)
        var handsOneColumn = handCount % 10
        val movingDown = handCount / 10 % 2 == 0

        if (movingDown) {
            handsOneColumn = 10 - (handsOneColumn - 1)
        } else {
            if (handsOneColumn == 0 || handsOneColumn == 1) {
                return 1
            }
        }

        return if (handsOneColumn > maxPossible) maxPossible else handsOneColumn
    }

    fun calculatePointsEarned(bid: Int, tricksTaken: Int): Int {
        return if (bid != tricksTaken) {
            tricksTaken - 5
        } else tricksTaken + 10
    }


}
